src/legacy/core_plugins/apm_oss/index.js

```diff
@@ -38,7 +38,10 @@ export default function apmOss(kibana) {
         spanIndices: Joi.string().default('apm-*'),
         metricsIndices: Joi.string().default('apm-*'),
         onboardingIndices: Joi.string().default('apm-*'),
-        apmAgentConfigurationIndex: Joi.string().default('.apm-agent-configuration')
+        apmAgentConfigurationIndex: Joi.string().default('.apm-agent-configuration'),
+
+        // UI settings
+        maxTraceItems: Joi.number().default(1000)
```

Comment:
I know this is super confusing but since this is only related to APM UI this setting should not live in the apm_oss package but in the xpack/apm package. 
The only reason we created the apm_oss package was so users can still use the APM dashboards even if they have APM UI disabled.